.github/workflows/ci.yml

```diff
@@ -23,21 +33,30 @@ jobs:
           rm -f /usr/local/bin/2to3 /usr/local/bin/2to3-3.11 /usr/local/bin/idle3 /usr/local/bin/idle3.11 /usr/local/bin/pydoc3 /usr/local/bin/pydoc3.11 /usr/local/bin/python3 /usr/local/bin/python3-config /usr/local/bin/python3.11 /usr/local/bin/python3.11-config
           brew install pkg-config ninja meson
 
-      - name: Install dependencies
-        env:
-          ARCHIVE_NAME: deps2_macos_native_dyn_libkiwix.tar.xz
+      - name: Install deps
+        shell: bash
         run: |
-          wget -O- https://tmp.kiwix.org/ci/${{env.ARCHIVE_NAME}} | tar -xJ -C ${{env.HOME}}
+          ARCHIVE_NAME=deps2_macos_${{matrix.target}}_libkiwix.tar.xz
+          wget -O- http://tmp.kiwix.org/ci/${ARCHIVE_NAME} | tar -xJ -C $HOME
 
-      - name: Compile source code
+      - name: Compile
         env:
           PKG_CONFIG_PATH: ${{env.HOME}}/BUILD_native_dyn/INSTALL/lib/pkgconfig
           CPPFLAGS: -I${{env.HOME}}/BUILD_native_dyn/INSTALL/include
+        shell: bash
         run: |
-          meson . build --default-library=shared -Db_coverage=true
+          MESON_OPTION="--default-library=shared -Db_coverage=true"
```

Comment:
I find it more maintainable to not use github contexts in `run`. I prefer to pass all values as env instead:

- easier to test your code locally. You just have to prepare your exports and then copy-paste.
- Github logs prints the env always so it's easy to spot any mistake or missing value
- It looks like *dependencies* for the command which is good for renames/cleanups